src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -135,6 +135,12 @@ void CoreWorkerDirectActorTaskSubmitter::ConnectActor(const ActorID &actor_id,
     return;
   }
 
+  if (queue->second.rpc_client &&
+      queue->second.rpc_client->Addr().worker_id() == address.worker_id()) {
```

Comment:
Check the IP and port as well? I know they are not necessary, but it makes the logic more clear.